sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -519,6 +519,8 @@ object FunctionRegistry {
     expressionBuilder("mode", ModeBuilder),
     expression[HllSketchAgg]("hll_sketch_agg"),
     expression[HllUnionAgg]("hll_union_agg"),
+    expression[Product]("product"),
+    expression[CollectTopK]("collect_top_k"),
```

Comment:
`collect_top_k` is a ML-specific expression for now, I think we should not expose it here.

Reply:
Moved this out.